lib/internal/fs/promises.js

```diff
@@ -577,7 +577,11 @@ async function read(handle, bufferOrParams, offset, length, position) {
     validateInteger(offset, 'offset', 0);
   }
 
-  length |= 0;
+  if (length == null) {
+    length = buffer.byteLength - offset;
+  } else {
+    length |= 0;
+  }
```

Comment:
Aside from `fh.read(buffer)`, this change will also affect `fh.read(buffer, offset, undefined, position)` and `fh.read(buffer, offset, null, position)`.
`position` and `length` in this signature are not optional, hence there's no default value.
How about adding `arguments.length` check instead?
Alternatively, we can consider a https://github.com/nodejs/node/labels/semver-major change making last positional arguments optional and aligning `fs.read()` and `fs.readSync()` accordingly.